components/Button/Button.tsx

```diff
@@ -0,0 +1,53 @@
+import Image from "next/image";
+import React, { MouseEventHandler } from "react";
+interface Appearance {
+  link: string;
+  buttonWithIcon: string;
+  button: string;
+}
+
+const APPEARANCE: Appearance = {
+  link: "md:rounded focus-within:shadow-md shadow-turquoise py-1 px-2 text-xl font-extrabold text-blue hover:text-orange  focus:text-blue  font-outfit underline",
+  buttonWithIcon:
+    "flex items-center   md:rounded focus-within:shadow-md shadow-turquoise py-1 px-2 text-xl font-extrabold text-blue hover:text-orange  focus:text-blue  font-outfit underline",
+  button:
+    "bg-orange hover:bg-blue focus:bg-orange  md:rounded  focus-within:shadow-md shadow-turquoise py-4 px-8 text-xl font-extrabold text-white font-outfit ",
+};
+
+type Appearances = keyof Appearance;
+
+interface IButtonProps
+  extends React.DetailedHTMLProps<
+    React.ButtonHTMLAttributes<HTMLButtonElement>,
+    HTMLButtonElement
+  > {
+  disabled?: boolean;
+  appearance: Appearances;
+  handleClick?: MouseEventHandler;
+  buttonAttributes?: Partial<HTMLButtonElement>;
+  icon?: string;
+  text: string;
+}
+
+export const Button: React.FC<IButtonProps> = (props: IButtonProps) => {
+  const { appearance, text, disabled, icon, handleClick, ...rest } = props;
+  const getAppearance = (appearance: Appearances): string => {
+    return APPEARANCE[appearance];
+  };
+
+  return (
+    <button
+      disabled={disabled}
+      className={getAppearance(appearance)}
```

Comment:
Me pregunto si deberiamos agregar un color para el estado `disabled`, que piensan?

Por otro lado, estaria bueno agregar `cursor-not-allowed` cuando el boton esta deshabilitado

Reply:
Tambien, estaria bueno separar los estilos "base" (los que se repiten) fuera de la coleccion `APPEARANCE`, por ej.: las fuentes... Asi si toca cambiarlas, se cambia en un solo lugar

Reply:
> Me pregunto si deberiamos agregar un color para el estado `disabled`, que piensan?
> 
> Por otro lado, estaria bueno agregar `cursor-not-allowed` cuando el boton esta deshabilitado

Lo de el color para el disabled creo que si estaria bueno, para que resiva feedback el usuario que no es un boton clickeable.

Reply:
ahora pusheo algo que es de tailwind para cuando este disabled `disabled:opacity-25` . lo subo y me dicen que onda si les gusta. tambien le agregue lo que dijiste @TonyMckes , el `cursor-not-allowed`.